source/agora/consensus/EnrollmentManager.d

```diff
@@ -331,6 +335,10 @@ public class EnrollmentManager
             this.preimages ~= hashFull(this.preimages[i]);
         this.data.random_seed = this.preimages[$-1];
 
+        // Reveal preimage in advance
+        this.data.preimage.round = this.data.cycle_length - PreimagePeriod;
```

Comment:
So the idea is, every 6 rounds the node will release a preimage for the next 6 rounds, right?

Therefore, this data needs to be updated. So shouldn't these two statements be inside of a function that can be called later to update and reveal the preimage?

Reply:
The data will be of course updated during revealing the pre-image after enrollment and the code will be added in another PR. Anyway, I try to apply what you said to the code by making another function for getting an pre-image of a certain round.